src/main/java/uk/gov/hmcts/ccd/domain/service/processor/CaseViewFieldProcessor.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewField;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewFieldBuilder;
+import uk.gov.hmcts.ccd.domain.model.definition.WizardPageField;
+import uk.gov.hmcts.ccd.domain.types.BaseType;
+
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.COLLECTION;
+import static uk.gov.hmcts.ccd.domain.model.definition.FieldType.COMPLEX;
+
+public abstract class CaseViewFieldProcessor extends FieldProcessor {
+
+    public CaseViewFieldProcessor(CaseViewFieldBuilder caseViewFieldBuilder) {
+        super(caseViewFieldBuilder);
+    }
+
+    public CaseViewField execute(CaseViewField caseViewField) {
+        final BaseType fieldType = BaseType.get(caseViewField.getFieldType().getType());
```

Comment:
we could reduce boilerplate by adding a 
isCollectionFieldType() and isComplexFieldType() methods to `FieldType`